src/App.tsx

```diff
@@ -15,61 +15,114 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-enum SortType {
-  NONE,
-  ALPABET,
-  LENGTH,
-}
-
-// Use this function in the render method
-function getReorderedGoods(
-  goods: string[],
-  sortType: SortType,
-  isReversed: boolean,
-) {
-  // Not to mutate the original array
-  const visibleGoods = [...goods];
-
-  // Sort and reverse goods if needed
-  // ...
-
-  return visibleGoods;
-}
-
-// DON'T save goods to the state
 type State = {
   isStarted: boolean,
   isReversed: boolean,
-  sortType: SortType,
+  sortType: string,
 };
 
-export const App = () => (
-  <div className="App">
-    <button type="button">
-      Start
-    </button>
+export class App extends Component<{}, State> {
+  state: State = {
+    isStarted: false,
+    isReversed: false,
+    sortType: '',
+  };
+
+  reverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  };
 
-    <button type="button">
-      Sort alphabetically
-    </button>
+  sortByABC = () => {
+    this.setState({ sortType: 'ALPHABET' });
+  };
 
-    <button type="button">
-      Sort by length
-    </button>
+  sortByLength = () => {
+    this.setState({ sortType: 'LENGTH' });
+  };
 
-    <button type="button">
-      Reverse
-    </button>
+  reset = () => {
+    this.setState({
+      isReversed: false,
+      sortType: '',
+    });
+  };
 
-    <button type="button">
-      Reset
-    </button>
+  render() {
+    const visibleGoods = [...goodsFromServer];
 
-    <ul className="Goods">
-      <li className="Goods__item">Dumplings</li>
-      <li className="Goods__item">Carrot</li>
-      <li className="Goods__item">Eggs</li>
-      <li className="Goods__item">...</li>
-    </ul>
-  </div>
-);
+    visibleGoods.sort((g1, g2) => {
+      switch (this.state.sortType) {
+        case 'ALPHABET':
+          return g1.localeCompare(g2);
+        case 'LENGTH':
+          return g1.length - g2.length;
+        default:
+          return 0;
+      }
+    });
+
+    if (this.state.isReversed) {
+      visibleGoods.reverse();
+    }
```

Comment:
This logic shouldn't be in the `render` method. Create a separate method for this. Method `render` should be as clear as possible.